proto-grpc.md

```diff
@@ -0,0 +1,254 @@
+
+This page is expected to continue as a 'live' document; as the ecosystem evolves and new rulesets
+are released, this document can be edited at
+[SIG-rules-authors](https://github.com/bazel-contrib/SIG-rules-authors/tree/gh-pages) to continue to
+reflect today's state.
+
+
+## The Basics
+
+At its most basic level, the task of working with Protobuf and gRPC `.proto` files is much the same
+as elsewhere: we execute the Protobuf Compiler `protoc` with the necessary arguments, environment
+and plugins to get it to produce a desired set of files or libraries for any specific language. For
+those unfamiliar with protoc, it is a small compiler executable that parses sets of `.proto`
+formatted files and executes plugins to write a set of output source code files. These files
+implement the interfaces for working with the messages and services in the language supported by the
+plugin.
+
+However, given the prevalence of Protobuf and gRPC use, users are not expected to reinvent the rules
+needed for calling protoc themselves. Instead, the common base dependency ruleset
+[bazelbuild/rules_proto](https://github.com/bazelbuild/rules_proto) provides the `proto_library`
+rule which can be used to establish libraries of `.proto` files and map their dependencies. For
+example, given the following dependency tree of `.proto` files, we could define a set of
+`proto_library` targets:
+
+![Tree of proto files](./assets/proto-tree.png)
+
+```starlark
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+proto_library(
+  name = "proto_a",
+  srcs = ["A.proto"],
+)
+
+proto_library(
+  name = "proto_b",
+  srcs = ["B.proto"],
+)
+
+proto_library(
+  name = "proto_c",
+  srcs = ["C.proto"],
+  deps = [":proto_a", ":proto_b"], # Provides the depended files when calling protoc
+)
+```
+
+Note that we could have grouped all of these `.proto` files under a single `proto_library` in this
+particular case. In real codebases these files may be under separate directories and managed by
+different groups or companies. The convention for splitting or grouping libraries varies by target
+language and is somewhat a matter of local taste. In general you should be consistent within your
+workspace, but fundamentally the process underneath works the same regardless.
+
+At this point we have defined the abstract collections of `.proto` files, but these targets alone
+do not provide the source code artifacts to interact with the messages and services within our code,
+since we have not given protoc any plugins or target languages. Therefore, we need to introduce
+rulesets that can consume these targets and provide the language-specific plugins and knowledge to
+produce libraries.
+
+
+## The Ruleset Options
+
+Within the Bazel ecosystem, there are three main types of rulesets to consider when using Protobuf
+or gRPC:
+
+- **Core Dependencies**: The [Protobuf](https://github.com/protocolbuffers/protobuf/) and
```

Comment:
I wouldn't call these core, since the other options don't depend on them.

Maybe "canonical" is a better term, as these are the official ones provided by the upstream authors